libflux/src/core/semantic/types.rs

```diff
@@ -153,16 +155,20 @@ pub enum Error {
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
-            Error::CannotUnify(t, with) => write!(f, "{} != {}", t, with),
-            Error::CannotConstrain(t, with) => write!(f, "{} is not {}", t, with),
+            Error::CannotUnify(expected, found) => write!(f, "{} != {}", found, expected),
+            Error::CannotConstrain(found, expected) => write!(f, "{} is not {}", found, expected),
             Error::OccursCheck(tv, ty) => write!(f, "{} != {} (recursive type)", tv, ty),
             Error::MissingLabel(a) => write!(f, "record is missing label {}", a),
+            Error::ExtraLabel(a) => write!(f, "found unexpected label {}", a),
             Error::CannotUnifyLabel(a, t, with) => {
                 write!(f, "{} != {} (record label {})", t, with, a)
             }
             Error::MissingArgument(x) => write!(f, "missing required argument {}", x),
+            Error::ExtraArgument(x) => write!(f, "found unexpected argument {}", x),
             Error::CannotUnifyArgument(x, e) => write!(f, "{} (argument {})", e, x),
-            Error::CannotUnifyReturn(t, with) => write!(f, "return type {} != {}", t, with),
+            Error::CannotUnifyReturn(expected, found) => {
+                write!(f, "return type {} != {}", found, expected)
```

Comment:
What about here? Should we replace the `!=` with a sentence explanation?

Reply:
Yes identifying which is the expected type and which is the actual type is helpful.